lib/util/index.js

```diff
@@ -3,6 +3,17 @@
 let path = require("path");
 let crypto = require("crypto");
 
+exports.requireOptional = (pkg, errorMessage, supplier = pkg) => {
```

Comment:
not entirely sure this name is correct

Reply:
It is not. I would say it is "requireOrRaise"?

Reply:
I don't like names like that. But since I don't currently have any better ideas: `requireOrPrompt` seems more accurate?

Reply:
Yep, agreed. I'm not a fan of these names. But if that's what it's doing, then I think the name should describe it.

Reply:
After letting this percolate for a while, I've concluded that it should be called `loadExtension`, because that is in fact the purpose here: Load a module which is an extension to (i.e. not a core part of and thus installed separately) whatever context we're currently in. (Not necessarily just faucet plugins.) Acceptable?

Reply:
Yes.